spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java

```diff
@@ -102,20 +105,27 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public Datastore datastore() {
```

Comment:
I'm worried that we breaking backwards compatibility because users won't be able to simply autowire a `Datastore` bean anymore. Can we conditionally create a `Datastore` bean if the default namespace provider is used?

Reply:
My thinking here was the user could append `() -> ` to the front of the `Datastore`  custom bean they had previously. 

But to your point, I think we could still accept a `Datastore` bean. But I don't think we can make any assumptions about the namespace provider since the user may totally ignore it or use it in his own Datastore object.

Reply:
Made change and added test. PTAL @meltsufin